### ISSUE TITLE
cibuildwheel GitHub Action creates a lot of spurious notifications

### ISSUE THREAD

**Issue by rgommers** [scipy/scipy#16970]
This is for regular opening of a PR, and I think it also happens when adding a label to a PR for example:

<img width="416" alt="image" src="https://user-images.githubusercontent.com/98330/188610957-db769c9d-c6b7-4fa0-92b9-bc992d7ce6b6.png">

The trigger conditions are unusual:
```yaml
on:
  schedule:
  #        ┌───────────── minute (0 - 59)
  #        │  ┌───────────── hour (0 - 23)
  #        │  │ ┌───────────── day of the month (1 - 31)
  #        │  │ │ ┌───────────── month (1 - 12 or JAN-DEC)
  #        │  │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)
  #        │  │ │ │ │
  - cron: "9  9 * * 6"
  # push:
  pull_request:
      types: [labeled, opened, synchronize, reopened]
  workflow_dispatch:
```

It may be those conditions, or it may be the `needs: get_commit_message` condition of `build_wheels`, which is also unique.

Cc @andyfaff, @lithomas1 in case one of you has already looked at this. scikit-learn has an alternative method: https://github.com/scikit-learn/scikit-learn/blob/main/.github/workflows/wheels.yml#L37. @thomasjpfan are you happy with how that functions?

**Comment by andyfaff** [comment]
It's not clear to me what you mean by spurious notifications. Are you getting sent emails  that the run is being skipped?

We don't necessarily need `labelled`. Numpy uses that when they add specific labels to a PR.
The other conditions are a subset of all the other pull_request possibilities for triggering a run (probably the only ones we need).

**Comment by rgommers** [comment]
Yes, I get emails with those notifications. I think others will too, when they open or merge a PR.

**Comment by andyfaff** [comment]
Ok, I've not seen any of those notifications. I guess they'd be very annoying if you got enough of them. I don't know why they get issued or whether you can stop them. I wonder why numpy doesn't experience the same issue.
 I guess the sklearn approach is just as good an alternative.

The conditions we want the wheels to be built are:
- if a commit message contains [wheel build]
- once a week (scheduled)
- if there is a release tag
- if it gets manually started (workflow dispatch)

I think the push on line 20 needs to be uncommented.

**Comment by andyfaff** [comment]
You might be getting notified because you merged the PR that contained the workflow.

**Comment by lithomas1** [comment]
It could be that since its a two stage pipeline, and the get-commit-message part always runs, but you still get notified when the wheel builders part get skipped, since part of it ran. (see for example, https://github.com/scipy/scipy/actions/runs/2999316305)

**Comment by thomasjpfan** [comment]
> @thomasjpfan are you happy with how that functions?

I am happy with it, but I have all GitHub notifications turned off for Actions.

scikit-learn has the same type of pipeline where the first step [always runs](https://github.com/scikit-learn/scikit-learn/actions/runs/3000749681). Based on the [pull_request event docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request), the only difference between SciPy and scikit-learn in terms of the `pull_request` event is the "labeled" trigger.

@rgommers Do you also get notified for skipped workflows from your fork such as the [Build GitHub Docker image](https://github.com/rgommers/scipy/actions/workflows/gitpod.yml) workflow?

**Comment by andyfaff** [comment]
I forgot, I do get notifications for PRs made against my fork, but not against the main repo

**Comment by rgommers** [comment]
> @rgommers Do you also get notified for skipped workflows from your fork such as the [Build GitHub Docker image](https://github.com/rgommers/scipy/actions/workflows/gitpod.yml) workflow?

No, pretty sure I don't. It's just this action. 



> I forgot, I do get notifications for PRs made against my fork, but not against the main repo

So it's either a "it's a PR you merged" thing or a permissions/owner thing.

Thanks for the feedback everyone, I'll try to figure it out through some changes to `wheels.yml`. And if I'm the only one who sees this then I may deal with it with an email filter.
